app/bundles/LeadBundle/Entity/LeadRepository.php

```diff
@@ -578,6 +578,34 @@ public function getEntities($args = array())
             ) : $results;
     }
 
+
+    public function getEntitiesLight($args = array()){
+    	$limit = 20000; //lines
+    	$results = array();
+    	$offset = 0;
+    	$result = $this->requestEntitiesLight($args, $offset, $limit);
+    	$results = array_merge($results, $result);
+    	while (count($result) === $limit){
+    		$offset += $limit;
+    		$result = $this->requestEntitiesLight($args, $offset+1, $limit);
+    		$results = array_merge($results, $result);
+    	}
+    	return $results;
+    }
```

Comment:
Doctrine has a built in way of handling this that we utilize in getEntities by passing $args['iterator_mode'] = true. https://github.com/mautic/mautic/blob/55d0d60fd3dc888ed946fbba46983a3a12a568c1/app/bundles/CoreBundle/Entity/CommonRepository.php#L131-L131

Example usage https://github.com/mautic/mautic/blob/d99c39ba0dd872167b552ef4454584538e1c396c/app/bundles/CampaignBundle/Command/UpdateLeadCampaignsCommand.php#L69-L69. 

Also, Doctrine will hold all it's entities in memory until released. Thus, if you know for a fact that the entity will no longer be used (such as exporting), detach it after use. I.e.

```
/** @var Mautic\LeadBundle\Entity\Lead $lead **/
$this->em->detach($lead); 
```

Or you can detach all entities of a specific type:
```
$this->em->clear('Mautic\LeadBundle\Entity\Lead');
```